api/src/main/java/edu/wpi/first/shuffleboard/api/plugin/Plugin.java

```diff
@@ -26,21 +28,73 @@
  * encounter order, which is alphabetical by jar name. For example, if a jar file "my_plugins.jar" defines a plugin
  * with ID "foo.bar" and another jar file "more_plugins.jar" <i>also</i> defines a plugin with that ID, the plugin
  * from "more_plugins" will be loaded first, then unloaded and replaced with the one from "my_plugins.jar". For this
- * reason, plugin authors should be careful to use unique group IDs. We recommend Java's reverse-DNS naming scheme.</p>
+ * reason, plugin authors should be careful to use unique group IDs. We recommend Java's reverse-DNS naming scheme.
+ *
+ * <p>Plugins <i>must</i> define their properties (developer group ID, name, current version, and summary) with a
+ * {@link Description @Description} annotation. Defining it in an annotation rather than an instance or class method or
+ * field allows shuffleboard to get data about the plugin without having to load the class or create a new instance of
+ * it. This approach prevents {@link NoClassDefFoundError NoClassDefFoundErrors} and
+ * {@link NoSuchMethodError NoSuchMethodErrors} that can arise if a plugin depends on another, or on classes defined in
+ * another plugin JAR, that has not been loaded or is not on the classpath.
+ *
+ * <p>For the same reasons, a plugin that depends on another <i>must</i> provide that information with a
+ * {@link Requires @Requires} or {@link Requirements @Requirements} annotation. The former is a <i>repeatable</i>
+ * annotation for which the latter is a wrapper; as such, it is recommended to use {@code @Requires} annotations to
+ * increase readability.
  */
 public class Plugin {
 
   private final String groupId;
   private final String name;
-  private final String version;
-  private final String description;
+  private final Version version;
+  private final String summary;
   private final BooleanProperty loaded = new SimpleBooleanProperty(this, "loaded", false);
 
-  protected Plugin(String groupId, String name, String version, String description) {
-    this.groupId = groupId;
-    this.name = name;
-    this.version = version;
-    this.description = description;
+  /**
+   * Creates a new plugin instance. The subclass <i>must</i> have a {@link Description @Description} annotation that
+   * defines the group ID, name, version, and summary of the plugin. If the plugin depends on another (eg the camera
+   * server plugin depends on the network tables one), that should be specified with a {@link Requires @Requires}
+   * annotation.
+   *
+   * @throws InvalidPluginDefinitionException if no {@code @Description} annotation is present, or if the
+   */
+  protected Plugin() {
+    validatePluginClass(getClass());
+    Description description = getClass().getAnnotation(Description.class);
+
+    this.groupId = description.group();
+    this.name = description.name();
+    this.version = Version.parse(description.version());
```

Comment:
I'd turn this into a type where all this stuff is combined into one thing. Eg. `PluginIdentifier` type. Seems like you are conflating this information with the `Plugin` type.

If you create one type `PluginIdentifier` you can then put the validation onto the `create` method for that type.